src/registry/artifactory.rs

```diff
@@ -1,36 +1,50 @@
 // (c) Copyright 2023 Helsing GmbH. All rights reserved.
 
-use std::sync::Arc;
-
 use eyre::{ensure, Context, ContextCompat};
-use serde::{Deserialize, Serialize};
 use url::Url;
 
 use super::Registry;
-use crate::{manifest::Dependency, package::Package};
+use crate::{
+    credentials::Credentials,
+    manifest::{Dependency, RegistryUri, Repository},
+    package::Package,
+};
 
 /// The registry implementation for artifactory
 #[derive(Debug, Clone)]
-pub struct Artifactory(Arc<ArtifactoryConfig>);
+pub struct Artifactory {
+    credentials: Credentials,
+    registry: RegistryUri,
+}
 
 impl Artifactory {
+    pub fn new(credentials: Credentials, registry: RegistryUri) -> Self {
+        Self {
+            credentials,
+            registry,
+        }
+    }
+
     /// Pings artifactory to ensure registry access is working
     pub async fn ping(&self) -> eyre::Result<()> {
-        let repositories_uri: Url = {
-            let mut url = self.0.url.clone();
-            url.set_path(&format!("{}/api/repositories", url.path()));
-            url
+        let token = self
+            .credentials
+            .get(&self.registry)
+            .wrap_err("no token found for the given registry")?;
```

Comment:
This is not an error – auth is optional!

Reply:
Good catch! Fixed!